blueocean-dashboard/src/main/js/creation/bitbucket/server/BbAddServerDialog.jsx

```diff
@@ -0,0 +1,192 @@
+
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            pending: false,
+            unknownError: null,
+            nameValue: null,
+            nameErrorMsg: null,
+            urlValue: null,
+            urlErrorMsg: null,
+        };
+
+        t = props.flowManager.translate;
+    }
+
+    _nameChange(value) {
+        this.setState({
+            nameValue: value,
+        });
+
+        this._updateNameErrorMsg();
+    }
+
+    _updateNameErrorMsg = debounce(() => {
+        if (this.state.nameErrorMsg && this.state.nameValue) {
+            this.setState({
+                nameErrorMsg: null,
+            });
+        }
+    }, 200);
+
+    _urlChange(value) {
+        this.setState({
+            urlValue: value,
+        });
+
+        this._updateUrlErrorMsg();
+    }
+
+    _updateUrlErrorMsg = debounce(() => {
+        if (this.state.urlErrorMsg && this.state.urlValue) {
+            this.setState({
+                urlErrorMsg: null,
+            });
+        }
+    }, 200);
+
+    _onCreateClick() {
+        const valid = this._performValidation();
+
+        if (!valid) {
+            return;
+        }
+
+        const { serverManager } = this.props.flowManager;
+
+        serverManager.createServer(this.state.nameValue, this.state.urlValue)
+            .then(
+                server => this._onCreateServerSuccess(server),
+                error => this._onCreateServerFailure(error),
+            );
+
+        this.setState({
+            pending: true,
+        });
+    }
+
+    _performValidation() {
+        let result = true;
+
+        if (!this.state.nameValue) {
+            this.setState({
+                nameErrorMsg: t('creation.bbserver.add_server.text_name_error_required'),
+            });
+
+            result = false;
+        }
+
+        if (!this.state.urlValue) {
+            this.setState({
+                urlErrorMsg: t('creation.bbserver.add_server.text_url_error_required'),
+            });
+
+            result = false;
+        }
+
+        return result;
+    }
+
+    _onCreateServerSuccess(server) {
+        this.setState({
+            pending: false,
+            unknownError: null,
+        });
+
+        this._onCloseClick(server);
+    }
+
+    _onCreateServerFailure(error) {
+        const { duplicateName, duplicateUrl, invalidUrl } = error;
+
+        const newState = {
+            pending: false,
+            unknownError: null,
+            nameErrorMsg: null,
+            urlErrorMsg: null,
+        };
+
+        if (duplicateName) {
+            newState.nameErrorMsg = t('creation.bbserver.add_server.text_name_error_duplicate');
+        }
+
+        if (duplicateUrl) {
+            newState.urlErrorMsg = t('creation.bbserver.add_server.text_url_error_duplicate');
+        } else if (invalidUrl && error.errors) {
+            error.errors.map(err => (
+                    newState.urlErrorMsg = err.message
```

Comment:
🐛  Adding a server with name and URL of "foo" results in bad request but the UI does not update with the error message. The `err` object only contains `code` and `field` properties, so `newState.urlErrorMsg` ends up being `undefined`. You might want to just to remove the `map` code and just assign `newState.urlErrorMsg = error.message`